### DISCUSSION TITLE
Components to freeze for YourTTS finetuning

### DISCUSSION THREAD

**Discussion by fColangelo** [coqui-ai/TTS#1308] [General Q&A]
Hi all. 

I am trying to finetune a YourTTS model that I trained. Specifically, I have added the Italian language with the M-AILABS dataset and would like to try finetuning with a small amount of data (~2 minutes). I have read that in this case is recommended to freeze the model, but the configs show multiple freezable components (freeze_encoder,  freeze_DP, freeze_PE, freeze_flow_decoder, freeze_waveform_decoder).  I think it would make sense to freeze encoder and decoder and duration predictor, but I have no clue concerning the other components. Could someone provide some guidance on the rationale for freezing the different components? Thanks!

**Comment by iamkhalidbashir** [comment]
Interested to know

**Comment by erogol** [comment]
@Edresson might inform us about that.

**Comment by Edresson** [comment]
> f data (~2 minutes). I have read that in this case is recommended to freeze the model, but the configs show multiple freezable components (freeze_encoder, freeze_DP, freeze_PE, freeze_flow_decoder, freeze_waveform_decoder). I think it would make sense to freeze enc

Hi @fColangelo ,

Frozen layers are not needed if you were fine-tuning the model using the whole dataset plus the new speaker (this approach is what we have done in YourTTS paper). However, in this case, you should guarantee that this speaker appears constantly on the batch.  In YourTTS paper we used Weighted random sampling to guarantee that samples from adapted speakers appear in a quarter of the batch. On the current TTS version, You could do that by using the code below, changing `speaker1` to the name of the new speaker and `number_of_speaker` by the number total of speakers in your dataset:

```
    #  compute the weight needed considering that each speaker has a default weight equal to 1
    adapted_speaker_weight = int(number_of_speaker/4)
    
    # Enable the weighted sampler (It should be added on VitsConfig)
    use_weighted_sampler=True,
    # Ensures that all speakers are seen in the training batch equally no matter how many samples each speaker has
    weighted_sampler_attrs={"speaker_name": 1.0},
    weighted_sampler_multipliers={
        "speaker_name": {
            "speaker1": adapted_speaker_weight}
    }

```

About the frozen parts. In general, freezing the encoder prevents the model from losing vocabulary. Freezing the flow-dec, vocoder, and posterior encoder to avoid the model losing speaker similarity.

**Reply by iamkhalidbashir** [reply]
So if we want to not use the original dataset and fine tune with just the yourtts model then we need to freeze just the encoder ?
